d2/src/main/java/com/linkedin/d2/balancer/properties/UriPropertiesJsonSerializer.java

```diff
@@ -109,6 +110,66 @@ public UriProperties fromBytes(byte[] bytes, long version) throws PropertySerial
     return uriProperties;
   }
 
+  public UriProperties fromProto(XdsD2.D2URI uri) throws PropertySerializationException
```

Comment:
ideally, we can directly use XdsD2.D2URI w/o constructing a new object UriProperties, however, I understand that this restli solution will be deprecated eventually, so it is not a priority to optimize this for now.

Reply:
Yeah to do so would imply refactoring pretty much the entirety of the D2 load balancing code to use the new class, along with the ZK monitoring code since it needs to translate the ZK JSON into the new proto D2URI class. Definitely not worth it at this stage